src/main/java/backend/UpdateController.java

```diff
@@ -61,6 +62,15 @@ public void processAndRefresh(List<FilterPanel> filterPanels) {
                     if (toUpdate.isEmpty()) {
                         // If no issues requiring metadata update, just run the filter and sort.
                         logic.updateUI(processFilters(filterExprs));
+                        ArrayList<CompletableFuture<Boolean>> stub = new ArrayList<>();
+                        Futures.sequence(stub)
+                                .thenRun(() -> {
+                                    if (isResumeTimer) {
+                                        logic.refreshTimer.resume();
+                                    }
+                                })
+                                .thenCompose(n -> logic.getRateLimitResetTime())
+                                .thenApply(logic::updateRateLimits);
```

Comment:
I think we shouldn't implement the `pause`, `resume` logic in this PR yet. This will introduce additional couplings and more openings for concurrency bugs. Adapting the refresh timer to the previous used API calls should be sufficient here.

We can have more careful planning and discussion in the future if it is necessary to implement this. But I think it's more important to keep things consistent even if the API quota runs out than trying to always have API quota. So some degree of efforts to be economical about it should be sufficient for now.